File: imageChangerApp/app/src/main/java/com/example/imagechangerapp/MainActivity.java
package com.example.imagechangerapp;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity{
    ImageView imageView;
    ArrayList<Integer> arrayList;
    Button next;
    Button prev;
    int i=0;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        imageView=findViewById(R.id.imageView2);
        next=findViewById(R.id.next);
        prev=findViewById(R.id.prev);
        arrayList=new ArrayList<>();
        arrayList.add(R.drawable.img1);
        arrayList.add(R.drawable.img2);
        arrayList.add(R.drawable.img3);
        arrayList.add(R.drawable.img4);
        arrayList.add(R.drawable.img5);

        next.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(i==4)
                i=0;
                else
                    i++;
                imageView.setImageResource(arrayList.get(i));
            }
        });
        prev.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(i==0)
                    i=4;
                else
                    i--;
                imageView.setImageResource(arrayList.get(i));
            }
        });
    }
}

File: RuntimePermission/app/src/main/java/com/example/runtimepermission/MainActivity.kt
package com.example.runtimepermission

import android.app.Activity
import android.os.Build
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat

class MainActivity : AppCompatActivity() {

    var PERMISSION_CODE = 100
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val permission = arrayOf<String>(
            "android.permission.ACCESS_BACKGROUND_LOCATION",
            "android.permission.ACCESS_MEDIA_LOCATION"
        )
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            requestPermissions(permission, PERMISSION_CODE)
        }

    }


    override fun onRequestPermissionsResult(
        requestCode: Int, permissions: Array<out String>,
        grantResults: IntArray
    ) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        if (permissions.contains("android.permission.ACCESS_BACKGROUND_LOCATION")
            && permissions.contains("android.permission.ACCESS_MEDIA_LOCATION")
        ) {
            Toast.makeText(
                applicationContext,
                "Media and background location access granted",
                Toast.LENGTH_SHORT
            ).show()

        } else if (permissions.contains("android.permission.ACCESS_BACKGROUND_LOCATION")) {
            Toast.makeText(
                applicationContext,
                "background location access granted",
                Toast.LENGTH_SHORT
            ).show()

        } else if (permissions.contains("android.permission.ACCESS_MEDIA_LOCATION")) {
            Toast.makeText(
                applicationContext,
                "media location access permission granted",
                Toast.LENGTH_SHORT
            ).show()
        }
    }
}
